src/index.js

```diff
@@ -1,29 +1,158 @@
 import './style.css';
 
-const listArray = [{
-  description: 'wash the dishes',
-  complete: true,
-  index: 0,
-},
-{
-  description: 'complet To Do list project',
-  complete: false,
-  index: 1,
-},
-];
+const textInput = document.querySelector('input');
+const listToDo = document.querySelector('.listToDo');
+const clearAllbtn = document.querySelector('button');
 
-const listToDo = document.getElementById('.myListToDo');
+class MyObject {
+  constructor(description, completed, index) {
+    this.description = description;
+    this.completed = completed;
+    this.index = index;
+  }
+}
+const myArray = [];
+const updateLocal = () => {
+  const localData = JSON.parse(localStorage.getItem('list'));
+  const toDos = document.querySelectorAll('span');
+  for (let i = 0; i < toDos.length; i += 1) {
+    if (toDos[i].classList.contains('checkToDo')) {
+      localData[i].completed = true;
+    } else {
+      localData[i].completed = false;
+    }
+  }
+  localStorage.setItem('list', JSON.stringify(localData));
+};
+const removeToDo = (toDo) => {
+  listToDo.removeChild(toDo);
+  const localData = JSON.parse(localStorage.getItem('list'));
+  const data = Array.from(localData).filter((i) => i.completed === false);
+  data.forEach((el, index) => {
+    el.index = index + 1;
+  });
+  localStorage.setItem('list', JSON.stringify(data));
+};
+const editToDo = (toDoContainer, toDo) => {
+  const editInput = document.createElement('input');
+  editInput.type = 'text';
+  editInput.className = 'editInput';
+  editInput.value = toDo.textContent;
+  toDoContainer.replaceChild(editInput, toDo);
+  editInput.addEventListener('keypress', (e) => {
+    if (e.key === 'Enter') {
+      const toDoContainers = document.querySelectorAll('.toDoContainer');
+      const localData = JSON.parse(localStorage.getItem('list'));
+      for (let i = 0; i < toDoContainers.length; i += 1) {
+        if (toDoContainers[i].classList.contains('checkedContainer')) {
+          localData[i].description = editInput.value;
+          localStorage.setItem('list', JSON.stringify(localData));
+        }
+      }
+      editInput.parentElement.classList.remove('checkedContainer');
+      toDoContainer.replaceChild(toDo, editInput);
+      toDo.textContent = editInput.value;
+    }
+  });
```

Comment:
- [ ] When I tap on a todo to edit it, I am able to get the input form to edit it. But when I tap outside, The todo that I had edited the details are not persisted in the local storage. Kindly look into this situation.